arch/arm/core/aarch32/cortex_m/vector_table_pad.ld

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2019 - 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Padding inserted after the (first-stage) vector table, so that the
+ * Zephyr image does not attempt to use the area which we reserve to
+ * detect null pointer dereferencing (0x0 - <size>). If the end of the
+ * vector table section is higher than the upper end of the reserved
+ * area, we add no padding.
+ *
+ * Note that even if the following linker script snippet is included
+ * multiple times, the padding will only be added at most once, to the
+ * first stage vector table.
+ */
+
+#if defined(CONFIG_CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION)
+. = MAX(ABSOLUTE(.), CONFIG_CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_PAGE_SIZE);
```

Comment:
it would be neat if CONFIG_CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_PAGE_SIZE could be set based on CONFIG_NUM_IRQS so there's no extra space used by default but I don't see an easy way to do that

Reply:
Good point, i'll give it a try, to see if there is a good way to do it. But actually, there's a tradeoff here: the larger the area, the more "protected" we are against null pointer dereferencing; there might be large arrays and dereferencing with 0 might fall outside this protected area if the area is small. So you might want to have a large area. x86 has 4kB, I am suggesting 1 kB here, as default :)